### PR TITLE
Build automation for performing common tasks

### PR DESCRIPTION
**Pull request by russcam**
Relates: #1051

This PR introduces a small F# project to automate common build tasks cross-platform:

- building all projects
- building agent zip file for use with startup hooks
- generating release notes
- building nuget packages

Tasks can be extended in future.

The root of the solution includes a `build.bat` and `build.sh` that serves as an entrypoint into invoking the project. For example

- to build all projects

    ```bat
    .\build.bat
    ```
   or
    ```bat
    ./build.sh
    ```

- to create a zip file for use with startup hooks

    ```bat
    .\build.bat agent-zip
    ```
   or
    ```bat
    ./build.sh agent-zip
    ```

etc.

The Targets.fs file in the F# scripts project defines all of the available targets/tasks and their dependencies, as well as command line arguments. Other .fs files define the functions to invoke for a given target/task.

This build automation may be of interest to the APM integration/automation team to automate tasks for the .NET agent on CI.

### CODE REVIEW

File: build.bat [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@echo Off
```
Comment (gregkalapos) [inline build.bat:1]: nit: could we add this and the `build.sh` file to the solution under `files` - we have all other files there. 

_I'd add a suggestion, but that part of the `.sln` file is unchanged._

**Review by gregkalapos — APPROVED** [review]
Awesome 🚀  This is really cool.

Just pinging @v1v: fyi we have some automation here that we could reuse later - some of this we already have, but some of them are new.

**Comment by v1v** [comment]
Just to be sure, this particular change is not tested in the CI at the moment, so do you want to enable it with a follow up? Or do you prefer as part of this PR?

**Comment by gregkalapos** [comment]
> Just to be sure, this particular change is not tested in the CI at the moment, so do you want to enable it with a follow up? Or do you prefer as part of this PR?

My thoughts on this: 
I think this is fine to merge as it is. This is part of https://github.com/elastic/apm-agent-dotnet/issues/1051, as a follow up I think it'd make sense to run `./build.sh agent-zip` in CI and make sure the generated `.zip` file ends up in jenkins artifacts. That way we 1) always build the zip file, so if we accidentally break something, likely the build will also fail, and 2) we could just take that file and publish it the way we want. There will be also another follow up to test the generated `.zip` file - and for that we'll also need to run `./build.sh agent-zip` in CI.

For the rest (like building NuGet packages, building the whole agent, etc.) I think there is no need to change those - for that I'd keep the current CI and just keep in mind that we have this and maybe reuse this later.

But this is @russcam's PR, so let's see what he says. :)

**Comment by russcam** [comment]
Agree with @gregkalapos, I think it'll make sense to run `./build.sh agent-zip` in CI, but it doesn't need to be part of this PR. I'll open an issue to track running in CI. 

Some more effort and investigation needs to go into testing the zip, so will open a separate issue for that too; it's related to what to do when different versions of `System.Diagnostics.DiagnosticSource` are used in an application versus the agent, and how this relates to subscribing to listeners in the startup hooks.

> For the rest (like building NuGet packages, building the whole agent, etc.) I think there is no need to change those - for that I'd keep the current CI and just keep in mind that we have this and maybe reuse this later.

I don't think there's an immediate need to change what currently exists in CI. I'd be happy to have a chat about using this build automation in CI in the future though; I've found it particularly useful in the past to be able to run the same commands locally as are run on CI, which this build automation helps with, without hopefully being overtly complex.

File: build.bat (hunk 1 above)
Comment (russcam) [reply build.bat:1]: Thanks, will add

File: .gitignore [hunk 2]
```diff
@@ -1,339 +1,343 @@
-## Ignore Visual Studio temporary files, build results, and
```
[the remark this answers is not in the document]
Comment (russcam) [reply .gitignore:1]: weird, I'll change it back

**Comment by russcam** [comment]
Opened https://github.com/elastic/apm-agent-dotnet/issues/1074 for the flaky test.

Merging this in.

**Pull request merged**